ingestion/src/metadata/data_quality/interface/test_suite_interface_factory.py

```diff
@@ -55,19 +42,18 @@ class TestSuiteInterfaceFactory:
 
     def __init__(self):
         """Initialize the interface factory"""
-        self._interface_type = {
-            "base": SQATestSuiteInterface,
-            DatalakeConnection.__name__: PandasTestSuiteInterface,
+        self._interface_type: Dict[str, Callable[[], Type[TestSuiteInterface]]] = {
+            "base": self.sqa,
         }
 
-    def register(self, interface_type: str, interface: TestSuiteInterface):
+    def register(self, interface_type: str, fn: Callable[[], Type[TestSuiteInterface]]):
```

Comment:
what's the typing `Callable[[], ...]`?

Reply:
😅 yep good question. It basically mean it is a callable `Callable[...]` that accepts no params (`[]`) and returns a `TestSuiteInterface` type (`Type[TestSuiteInterface]`).

Reply:
So it can be a function, a method, a class with a __call__() method, etc.